File: packages/shared/src/tax/countries.ts
/**
 * Country reference data: ISO 3166-1 alpha-2 code, display name, ISO 4217
 * currency, and the country's default standard VAT/GST/Sales-tax rate (%)
 * and label. Selecting a country during onboarding/firm-creation seeds the
 * tenant's `currency` and `vat_rate` from this table; both remain editable
 * afterwards in Settings.
 *
 * Mirrored in `apps/api/app/tax/countries.py` — keep the two in sync.
 */
export interface CountryInfo {
  code: string;
  name: string;
  currency: string;
  vatRate: number;
  vatName: string;
}

export const COUNTRIES: CountryInfo[] = [
  { code: "AE", name: "United Arab Emirates", currency: "AED", vatRate: 5, vatName: "VAT" },
  { code: "SA", name: "Saudi Arabia", currency: "SAR", vatRate: 15, vatName: "VAT" },
  { code: "PK", name: "Pakistan", currency: "PKR", vatRate: 18, vatName: "Sales Tax" },
  { code: "IN", name: "India", currency: "INR", vatRate: 18, vatName: "GST" },
  { code: "US", name: "United States", currency: "USD", vatRate: 0, vatName: "Sales Tax" },
  { code: "GB", name: "United Kingdom", currency: "GBP", vatRate: 20, vatName: "VAT" },
  { code: "CA", name: "Canada", currency: "CAD", vatRate: 5, vatName: "GST" },
  { code: "AU", name: "Australia", currency: "AUD", vatRate: 10, vatName: "GST" },
  { code: "DE", name: "Germany", currency: "EUR", vatRate: 19, vatName: "VAT" },
  { code: "FR", name: "France", currency: "EUR", vatRate: 20, vatName: "VAT" },
  { code: "IT", name: "Italy", currency: "EUR", vatRate: 22, vatName: "VAT" },
  { code: "ES", name: "Spain", currency: "EUR", vatRate: 21, vatName: "VAT" },
  { code: "NL", name: "Netherlands", currency: "EUR", vatRate: 21, vatName: "VAT" },
  { code: "BE", name: "Belgium", currency: "EUR", vatRate: 21, vatName: "VAT" },
  { code: "CH", name: "Switzerland", currency: "CHF", vatRate: 8.1, vatName: "VAT" },
  { code: "SE", name: "Sweden", currency: "SEK", vatRate: 25, vatName: "VAT" },
  { code: "NO", name: "Norway", currency: "NOK", vatRate: 25, vatName: "VAT" },
  { code: "DK", name: "Denmark", currency: "DKK", vatRate: 25, vatName: "VAT" },
  { code: "FI", name: "Finland", currency: "EUR", vatRate: 24, vatName: "VAT" },
  { code: "PL", name: "Poland", currency: "PLN", vatRate: 23, vatName: "VAT" },
  { code: "PT", name: "Portugal", currency: "EUR", vatRate: 23, vatName: "VAT" },
  { code: "IE", name: "Ireland", currency: "EUR", vatRate: 23, vatName: "VAT" },
  { code: "AT", name: "Austria", currency: "EUR", vatRate: 20, vatName: "VAT" },
  { code: "GR", name: "Greece", currency: "EUR", vatRate: 24, vatName: "VAT" },
  { code: "TR", name: "Turkey", currency: "TRY", vatRate: 20, vatName: "VAT" },
  { code: "EG", name: "Egypt", currency: "EGP", vatRate: 14, vatName: "VAT" },
  { code: "QA", name: "Qatar", currency: "QAR", vatRate: 0, vatName: "VAT" },
  { code: "KW", name: "Kuwait", currency: "KWD", vatRate: 0, vatName: "VAT" },
  { code: "BH", name: "Bahrain", currency: "BHD", vatRate: 10, vatName: "VAT" },
  { code: "OM", name: "Oman", currency: "OMR", vatRate: 5, vatName: "VAT" },
  { code: "JO", name: "Jordan", currency: "JOD", vatRate: 16, vatName: "Sales Tax" },
  { code: "LB", name: "Lebanon", currency: "LBP", vatRate: 11, vatName: "VAT" },
  { code: "CN", name: "China", currency: "CNY", vatRate: 13, vatName: "VAT" },
  { code: "JP", name: "Japan", currency: "JPY", vatRate: 10, vatName: "Consumption Tax" },
  { code: "KR", name: "South Korea", currency: "KRW", vatRate: 10, vatName: "VAT" },
  { code: "SG", name: "Singapore", currency: "SGD", vatRate: 9, vatName: "GST" },
  { code: "MY", name: "Malaysia", currency: "MYR", vatRate: 6, vatName: "Sales & Service Tax" },
  { code: "ID", name: "Indonesia", currency: "IDR", vatRate: 11, vatName: "VAT" },
  { code: "TH", name: "Thailand", currency: "THB", vatRate: 7, vatName: "VAT" },
  { code: "PH", name: "Philippines", currency: "PHP", vatRate: 12, vatName: "VAT" },
  { code: "VN", name: "Vietnam", currency: "VND", vatRate: 10, vatName: "VAT" },
  { code: "BD", name: "Bangladesh", currency: "BDT", vatRate: 15, vatName: "VAT" },
  { code: "NZ", name: "New Zealand", currency: "NZD", vatRate: 15, vatName: "GST" },
  { code: "ZA", name: "South Africa", currency: "ZAR", vatRate: 15, vatName: "VAT" },
  { code: "NG", name: "Nigeria", currency: "NGN", vatRate: 7.5, vatName: "VAT" },
  { code: "KE", name: "Kenya", currency: "KES", vatRate: 16, vatName: "VAT" },
  { code: "MA", name: "Morocco", currency: "MAD", vatRate: 20, vatName: "VAT" },
  { code: "TN", name: "Tunisia", currency: "TND", vatRate: 19, vatName: "VAT" },
  { code: "MX", name: "Mexico", currency: "MXN", vatRate: 16, vatName: "VAT" },
  { code: "IL", name: "Israel", currency: "ILS", vatRate: 17, vatName: "VAT" },
];

const COUNTRY_BY_CODE: Record<string, CountryInfo> = {};
for (const c of COUNTRIES) {
  COUNTRY_BY_CODE[c.code] = c;
}

export function getCountryInfo(code: string): CountryInfo | undefined {
  return COUNTRY_BY_CODE[code];
}

File: apps/api/app/tax/countries.py
"""Country reference data — mirrors packages/shared/src/tax/countries.ts.

Each entry provides the ISO 3166-1 alpha-2 code, display name, ISO 4217
currency, and the country's default standard VAT/GST/Sales-tax rate (%) and
label. Selecting a country during onboarding/firm-creation seeds the
tenant's `currency` and `vat_rate` from this table; both remain editable
afterwards in Settings.
"""

from dataclasses import dataclass


@dataclass(frozen=True)
class CountryInfo:
    code: str
    name: str
    currency: str
    vat_rate: float
    vat_name: str


COUNTRIES: list[CountryInfo] = [
    CountryInfo("AE", "United Arab Emirates", "AED", 5, "VAT"),
    CountryInfo("SA", "Saudi Arabia", "SAR", 15, "VAT"),
    CountryInfo("PK", "Pakistan", "PKR", 18, "Sales Tax"),
    CountryInfo("IN", "India", "INR", 18, "GST"),
    CountryInfo("US", "United States", "USD", 0, "Sales Tax"),
    CountryInfo("GB", "United Kingdom", "GBP", 20, "VAT"),
    CountryInfo("CA", "Canada", "CAD", 5, "GST"),
    CountryInfo("AU", "Australia", "AUD", 10, "GST"),
    CountryInfo("DE", "Germany", "EUR", 19, "VAT"),
    CountryInfo("FR", "France", "EUR", 20, "VAT"),
    CountryInfo("IT", "Italy", "EUR", 22, "VAT"),
    CountryInfo("ES", "Spain", "EUR", 21, "VAT"),
    CountryInfo("NL", "Netherlands", "EUR", 21, "VAT"),
    CountryInfo("BE", "Belgium", "EUR", 21, "VAT"),
    CountryInfo("CH", "Switzerland", "CHF", 8.1, "VAT"),
    CountryInfo("SE", "Sweden", "SEK", 25, "VAT"),
    CountryInfo("NO", "Norway", "NOK", 25, "VAT"),
    CountryInfo("DK", "Denmark", "DKK", 25, "VAT"),
    CountryInfo("FI", "Finland", "EUR", 24, "VAT"),
    CountryInfo("PL", "Poland", "PLN", 23, "VAT"),
    CountryInfo("PT", "Portugal", "EUR", 23, "VAT"),
    CountryInfo("IE", "Ireland", "EUR", 23, "VAT"),
    CountryInfo("AT", "Austria", "EUR", 20, "VAT"),
    CountryInfo("GR", "Greece", "EUR", 24, "VAT"),
    CountryInfo("TR", "Turkey", "TRY", 20, "VAT"),
    CountryInfo("EG", "Egypt", "EGP", 14, "VAT"),
    CountryInfo("QA", "Qatar", "QAR", 0, "VAT"),
    CountryInfo("KW", "Kuwait", "KWD", 0, "VAT"),
    CountryInfo("BH", "Bahrain", "BHD", 10, "VAT"),
    CountryInfo("OM", "Oman", "OMR", 5, "VAT"),
    CountryInfo("JO", "Jordan", "JOD", 16, "Sales Tax"),
    CountryInfo("LB", "Lebanon", "LBP", 11, "VAT"),
    CountryInfo("CN", "China", "CNY", 13, "VAT"),
    CountryInfo("JP", "Japan", "JPY", 10, "Consumption Tax"),
    CountryInfo("KR", "South Korea", "KRW", 10, "VAT"),
    CountryInfo("SG", "Singapore", "SGD", 9, "GST"),
    CountryInfo("MY", "Malaysia", "MYR", 6, "Sales & Service Tax"),
    CountryInfo("ID", "Indonesia", "IDR", 11, "VAT"),
    CountryInfo("TH", "Thailand", "THB", 7, "VAT"),
    CountryInfo("PH", "Philippines", "PHP", 12, "VAT"),
    CountryInfo("VN", "Vietnam", "VND", 10, "VAT"),
    CountryInfo("BD", "Bangladesh", "BDT", 15, "VAT"),
    CountryInfo("NZ", "New Zealand", "NZD", 15, "GST"),
    CountryInfo("ZA", "South Africa", "ZAR", 15, "VAT"),
    CountryInfo("NG", "Nigeria", "NGN", 7.5, "VAT"),
    CountryInfo("KE", "Kenya", "KES", 16, "VAT"),
    CountryInfo("MA", "Morocco", "MAD", 20, "VAT"),
    CountryInfo("TN", "Tunisia", "TND", 19, "VAT"),
    CountryInfo("MX", "Mexico", "MXN", 16, "VAT"),
    CountryInfo("IL", "Israel", "ILS", 17, "VAT"),
]

COUNTRY_BY_CODE: dict[str, CountryInfo] = {c.code: c for c in COUNTRIES}


def get_country_info(code: str) -> CountryInfo | None:
    return COUNTRY_BY_CODE.get(code)
